### PR TITLE
Fix Baton Pass and Teleport not failing with no switch-ins left

### PR DESCRIPTION
**Pull request by Karthik99999**

### CODE REVIEW

**Comment by urkerab** [comment]
Shouldn't this fail in doubles when you only have two left? (Similarly for triples in Gen 5/6. And don't get me started on rotation battles, but fortunately they're not supported yet.)

**Comment by DaWoblefet** [comment]
It should - can we leverage ``this.battle.canSwitch``?

**Comment by urkerab** [comment]
Sounds good to me - it's even compatible with PR #8677!

File: data/moves.ts [hunk 1]
```diff
@@ -1006,6 +1006,9 @@ export const Moves: {[moveid: string]: MoveData} = {
 		pp: 40,
 		priority: 0,
 		flags: {},
+		onTry(source) {
```
[the remark this answers is not in the document]
Comment (DaWoblefet) [reply data/moves.ts:1009]: Tested this on cart in Gen 8, and ``onTry`` is correct for Teleport (and it doubles Stomping Tantrum on failure). It won't trigger Libero but does deduct PP. Baton Pass, Lunar Dance, and Healing Wish should be using one of our later events in the checks before move success (they don't double Stomping Tantrum on failure). They happen sometime after Libero.

**Comment by DaWoblefet** [comment]
Thanks Karthik!

**Pull request merged**